### ISSUE TITLE
[iOS][Fallback] Custom Renderer fallback handling fails 

### ISSUE THREAD

**Issue by Aditya0603** [microsoft/AdaptiveCards#3718]
# Platform
* iOS

# Author or host
Host (Teams)

# Version of SDK
1.2.5

# Details
Fallback rendering for custom renderer fails while fallback element is text.

Steps - 
1/ create a custom renderer for ACRMedia
2/ Throw a ACOFallbackException in render

try the below JSON

{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.1",
  "fallbackText": "This card requires Media to be viewed. Ask your platform to update to Adaptive Cards v1.1 for this and more!",
  "body": [
    {
      "type": "Media",
      "poster": "https://adaptivecards.io/content/poster-video.png",
      "sources": [
        {
          "mimeType": "video/mp4",
          "url": "https://adaptivecardsblob.blob.core.windows.net/assets/AdaptiveCardsOverviewVideo.mp4"
        }
      ],
      "fallback": {
        "type": "FactSet",
        "facts": [
          {
            "title": "Fact 1",
            "value": "Value 1"
          },
          {
            "title": "Fact 2",
            "value": "Value 2"
          }
        ]
      }
    }
  ]
}

**Comment by shalinijoshi19** [comment]
@nesalang can you take a look here? Thanks for reporting @Aditya0603 / @VikrantSingh01

**Comment by nesalang** [comment]
@Aditya0603 ,

there are two apis to register custom renderers
1. setBaseCardElementRenderer 
Used to override the behavior of the renderers that renders known elements.
This is the one that should be used for this issue since Media type is a known adaptive cards element.
2. setCustomElementRenderer
Used to add a new type that is currently unknown to adaptive cards scheme, and it requires to register a corresponding parser that knows how to parse the custom type.

FallBack is thrown specifically, and it's due to the lack of custom parser in this case. Please make sure that correct api is used. 

Thank you.

**Comment by Aditya0603** [comment]
@nesalang I am using this method itself setBaseCardElementRenderer.

Let me rephrase the question here. 
For some specific case we don't need to render the Media element, and instead render the fallback element (if provided).

So I set a new renderer for Media using setBaseCardElementRenderer.

`[registration setBaseCardElementRenderer:[CustomMediaRenderer getInstance]
                                 cardElementType:ACRMedia];`
and threw a exception in the render method so that the fallback elements render

`- (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
          rootView:(ACRView *)rootView
            inputs:(NSArray *)inputs
   baseCardElement:(ACOBaseCardElement *)acoElem
        hostConfig:(ACOHostConfig *)acoConfig
{
    @throw [ACOFallbackException fallbackException];
}`

this crashed the app , instead it should have rendered the fallback elements

**Comment by nesalang** [comment]
@Aditya0603 thanks for the clarification. This is due to the text preprocessing for performance. I'm going to update the issue description.
